redis/connection.py

```diff
@@ -577,7 +605,12 @@ def connect(self):
 
         self._sock = sock
         try:
-            self.on_connect()
+            if self.redis_connect_func is not None:
```

Comment:
Since the majority/default case is that the redis_connect_func is None, we should prioritize the default connection, and instead call the func only if defined.

Reply:
Changed.